### PR TITLE
Moves crio CI to prow

### PR DESCRIPTION
**Pull request by wgahnagl**
adds prow configuration for running cri-o's CI tests

### CODE REVIEW

File: ci-operator/config/cri-o/test/cri-o-test-main.yaml
Comment (osherdp) [inline ci-operator/config/cri-o/test/cri-o-test-main.yaml:35]:
```suggestion
    cluster_profile: equinix-ocp-metal
```

otherwise it's billing Ecosystem Engineering organization :sweat_smile:

File: ci-operator/config/cri-o/test/cri-o-test-main.yaml
Comment (osherdp) [inline ci-operator/config/cri-o/test/cri-o-test-main.yaml:5]:
```suggestion
    tag: "8"
  dev-scripts:
    name: test
    namespace: ocp-kni
    tag: dev-scripts
```

The ``baremetalds-packet-setup`` step requires this image to be "imported", otherwise it will not work

File: ci-operator/step-registry/crio-test/agent/test/crio-test-agent-test-ref.yaml
Comment (osherdp) [inline ci-operator/step-registry/crio-test/agent/test/crio-test-agent-test-ref.yaml:3]: I guess you meant here:
```suggestion
  from: crio-test-agent-src
```

Otherwise you'll need to import ``assisted-installer-agent-src`` in ``base_images`` stanza

**Review by osherdp — CHANGES_REQUESTED** [review]

File: ci-operator/config/cri-o/test/cri-o-test-main.yaml
Comment (wgahnagl) [reply ci-operator/config/cri-o/test/cri-o-test-main.yaml:35]: oops!

File: ci-operator/jobs/cri-o/test/cri-o-test-main-presubmits.yaml
Comment (haircommander) [inline ci-operator/jobs/cri-o/test/cri-o-test-main-presubmits.yaml:15]: where did this profile come from? is it predefined

Comment (wgahnagl) [reply ci-operator/jobs/cri-o/test/cri-o-test-main-presubmits.yaml:15]: it is predefined :ok_hand: it's used in a lot of other places but I'm not sure where it's actually created

File: ci-operator/config/cri-o/test/cri-o-test-main.yaml
Comment (vrutkovs) [reply ci-operator/config/cri-o/test/cri-o-test-main.yaml:28]: `src` is being autocreated by ci-operator

File: ci-operator/step-registry/crio-test/agent/setup/crio-test-agent-setup.metadata.json
Comment (osherdp) [inline ci-operator/step-registry/crio-test/agent/setup/crio-test-agent-setup.metadata.json:2]: this is weird, I'm pretty surprised CI didn't yell about this path being unsynced with the file location

File: ci-operator/step-registry/crio-test/e2e/test/crio-test-e2e-test-commands.sh
Comment (rphillips) [inline ci-operator/step-registry/crio-test/e2e/test/crio-test-e2e-test-commands.sh:43]: this doesn't look right

File: ci-operator/jobs/cri-o/cri-o/cri-o-cri-o-main-presubmits.yaml
Comment (haircommander) [inline ci-operator/jobs/cri-o/cri-o/cri-o-cri-o-main-presubmits.yaml:25]: can this name be `ci/prow/critest-el8` to match our other tests

File: ci-operator/jobs/cri-o/cri-o/cri-o-cri-o-main-presubmits.yaml
Comment (haircommander) [inline ci-operator/jobs/cri-o/cri-o/cri-o-cri-o-main-presubmits.yaml:36]: if we do the above, then this should be `/test critest-el8`

File: ci-operator/jobs/cri-o/cri-o/cri-o-cri-o-main-presubmits.yaml
Comment (haircommander) [inline ci-operator/jobs/cri-o/cri-o/cri-o-cri-o-main-presubmits.yaml:228]: same naming scheme as above

File: ci-operator/step-registry/crio-test/OWNERS
Comment (haircommander) [inline ci-operator/step-registry/crio-test/OWNERS:2]: we don't need this anymore do we?

Comment (haircommander) [reply ci-operator/step-registry/crio-test/OWNERS:2]: yeah all of the crio-test stuff can go away methinks

File: ci-operator/step-registry/crio-test/critest/gather/crio-test-critest-gather-commands.sh
Comment (haircommander) [inline ci-operator/step-registry/crio-test/critest/gather/crio-test-critest-gather-commands.sh:1]: should this live in `ci-operator/step-registry/crio` instead of `crio-test`?

File: core-services/prow/02_config/_config.yaml
Comment (haircommander) [inline core-services/prow/02_config/_config.yaml:19]: what're these changes?

File: core-services/prow/02_config/_plugins.yaml
Comment (haircommander) [inline core-services/prow/02_config/_plugins.yaml:53]: we don't need ci-v2 anymore right? bc we're using main again

**Review by haircommander — COMMENTED** [review]
let's make sure we remove evidence of `crio-test` and `ci-v2`, as we only care abot `cri-o/cri-o:main` here

File: ci-operator/step-registry/crio-test/e2e/test/crio-test-e2e-test-commands.sh
Comment (rphillips) [inline ci-operator/step-registry/crio-test/e2e/test/crio-test-e2e-test-commands.sh:14]: need to remove the cat here

File: ci-operator/step-registry/crio-test/OWNERS
Comment (wgahnagl) [reply ci-operator/step-registry/crio-test/OWNERS:2]: the crio-test stuff is where all the tests are, actually lol 
should we rename it to just crio?

File: ci-operator/jobs/cri-o/cri-o/cri-o-cri-o-main-presubmits.yaml
Comment (haircommander) [reply ci-operator/jobs/cri-o/cri-o/cri-o-cri-o-main-presubmits.yaml:228]: el8 is shorthand for enterprise linux 8. It encompasses rhel and rebuilds (in this case rocky).

File: core-services/prow/02_config/_config.yaml
Comment (wgahnagl) [reply core-services/prow/02_config/_config.yaml:19]: it looks like it's just been moved from an autogeneration 

https://github.com/openshift/release/pull/28107/files/97d6a7e9fdb7a71e9fef93a7d38295222838fcef#diff-3fe4522c3fa00ecd918a986bc31b4cde479c02bcace2b0497bd136760edbd71fL37

**Comment by haircommander** [comment]
/approve

@rphillips LGTY?

File: ci-operator/step-registry/crio-test/e2e/test/crio-test-e2e-test-commands.sh
Comment (wgahnagl) [reply ci-operator/step-registry/crio-test/e2e/test/crio-test-e2e-test-commands.sh]: probably shouldn't have that there at all lol

File: ci-operator/jobs/cri-o/cri-o/cri-o-cri-o-main-presubmits.yaml
Comment (haircommander) [inline ci-operator/jobs/cri-o/cri-o/cri-o-cri-o-main-presubmits.yaml:25]: I still think this hsould be `critest-el8`, but am willing to let that be a follow up if needed

File: ci-operator/step-registry/cri-o/critest/test/cri-o-critest-test-commands.sh
Comment (sohankunkerkar) [inline ci-operator/step-registry/cri-o/critest/test/cri-o-critest-test-commands.sh:14]: Neat: we don't need this anymore

File: ci-operator/step-registry/cri-o/critest/test/cri-o-critest-test-commands.sh
Comment (sohankunkerkar) [inline ci-operator/step-registry/cri-o/critest/test/cri-o-critest-test-commands.sh:42]: We're already setting the GOPATH explicitly in `cri-o/test`, so no need to provide that argument here.

Comment (sohankunkerkar) [reply ci-operator/step-registry/cri-o/critest/test/cri-o-critest-test-commands.sh:42]: same for the other two tests as well

File: ci-operator/config/cri-o/cri-o/cri-o-cri-o-main__ci.yaml
Comment (sohankunkerkar) [inline ci-operator/config/cri-o/cri-o/cri-o-cri-o-main__ci.yaml:23]: I'm not sure if this stanza is actually building anything here.

File: ci-operator/step-registry/cri-o/e2e/test/OWNERS
Comment (rphillips) [inline ci-operator/step-registry/cri-o/e2e/test/OWNERS:2]: perhaps add peter,  mrunal, myself, and sohan as approvers

File: core-services/prow/02_config/cri-o/test/_pluginconfig.yaml
Comment (rphillips) [inline core-services/prow/02_config/cri-o/test/_pluginconfig.yaml:4]: we can remove cri-o/test module

File: core-services/prow/02_config/_config.yaml
Comment (rphillips) [inline core-services/prow/02_config/_config.yaml:14]: why is this block changed?

File: ci-operator/config/cri-o/cri-o/cri-o-cri-o-main__ci.yaml
Comment (rphillips) [reply ci-operator/config/cri-o/cri-o/cri-o-cri-o-main__ci.yaml:23]: this is correct... it moves the code into an image.

**Comment by jmguzik** [comment]
@haircommander Please notify me once you think your PR is ready. I'll approve. That being said you have still some things to fix I guess.

**Review by jmguzik — APPROVED** [review]

**Comment by jmguzik** [comment]
/hold
Please release when you think it's ready

**Pull request merged**
